qubesmanager/settings.py

```diff
@@ -882,6 +892,69 @@ def __init_advanced_tab__(self):
+                    self.appvm_default_bootmode_desc.setVisible(True)
+                    self.appvm_default_bootmode.setVisible(True)
+                else:
+                    self.appvm_default_bootmode_desc.setVisible(False)
+                    self.appvm_default_bootmode.setVisible(False)
+                self.bootmode_ids = [
+                    x.split('.')[2] for x in self.vm.features \
+                        if x.startswith("boot-mode.kernelopts.")
+                ]
+                subject = self.vm
+                while hasattr(subject, "template"):
+                    self.bootmode_ids.extend([
+                        x.split('.')[2] for x in subject.template.features \
+                            if x.startswith("boot-mode.kernelopts.")
+                    ])
+                    subject = subject.template
+                self.bootmode_names = [
+                    self.vm.features.check_with_template(
+                        f"boot-mode.name.{x}", x
+                    ) for x in self.bootmode_ids
+                ]
+                bootmode_widget_data = list(zip(
+                    self.bootmode_names, self.bootmode_ids
+                ))
+                bootmode_widget_data.sort()
+                utils.initialize_widget_for_property(
+                    widget=self.bootmode,
+                    choices=bootmode_widget_data,
+                    property_name="bootmode",
+                    holder=self.vm,
+                    allow_default=True,
+                    default_text_provider=get_default_bootmode_name
+                )
+                if hasattr(self.vm, "appvm_default_bootmode"):
+                    # We need to recreate the bootmode_widget_data list,
+                    # because utils.initialize_widget_for_property adds a
+                    # default item to the list as a side-effect, and we'd end
+                    # up with duplicate "default" entries if we didn't
+                    # reinitialize it. Modifying
+                    # initialize_widget_for_property to operate on a list deep
+                    # copy breaks the virtualization mode combo box, making
+                    # the default mode appear as
+                    # `<object object at 0xaddress>`.
+                    bootmode_widget_data = list(zip(
+                        self.bootmode_names, self.bootmode_ids
+                    ))
+                    bootmode_widget_data.sort()
+                    utils.initialize_widget_for_property(
+                        widget=self.appvm_default_bootmode,
+                        choices=bootmode_widget_data,
+                        property_name="appvm_default_bootmode",
+                        holder=self.vm,
+                        allow_default=True,
+                        default_text_provider=get_default_bootmode_name
+                    )
+                self.bootmode_kernel_opts.setText(
```

Comment:
From the other PR - can you add debug print to see `self.vm.bootmode` value here?

Reply:
I did that and saw that it was printed properly. I was able to fix this issue though using your event handling trick, I just ended up having to use a different event than the one you initially suggested. (I'm not entirely sure `feature-set` exists in the context of qubesvm.py, it seems only `domain-feature-set` does.)